django/thunderstore/modpacks/tasks.py

```diff
@@ -0,0 +1,50 @@
+import base64
+import io
+import zipfile
+import yaml
+
+from celery import shared_task
+from typing import Dict
+from thunderstore.core.settings import CeleryQueues
+from thunderstore.modpacks.models.legacyprofile import LegacyProfile, LegacyProfileMetaData
+
+def parse_package_name_and_team(s: str):
+    reversed_splitted = "".join(reversed(s)).split("-", maxsplit=1)
+    name = "".join(reversed(reversed_splitted[0]))
+    team = "".join(reversed(reversed_splitted[1]))
+    return (name, team)
+
+
+def generate_mod_metadata(mod: Dict):
+    name, team = parse_package_name_and_team(mod["name"])
+    version = ("%s.%s.%s" % (mod["version"]["major"], mod["version"]["minor"], mod["version"]["patch"]))
+    return {
+        "name": name,
+        "team": team,
+        "version": version,
+        "enabled": mod["enabled"],
+        "url": ("https://thunderstore.io/c/valheim/p/%s/%s/v/%s" % (
+            team,
+            name,
+            version
+        ))
+    }
+
+@shared_task(queue=CeleryQueues.Default)
```

Comment:
Generally the best practice with celery tasks is to have the actual logic for handling the task in its own module, and simply import & call it from the task.